### PR TITLE
RE-1842 HoldStep

### PR DESCRIPTION
**Pull request by hughsaunders**
Adds a hold step to the Nodepool Agents plugin, this lets
a user configure a freestle or pipeline job to hold the
current node from within a job.

Example:
    node("nodepool-debian"){
        nodePoolHold("testing")
    }

### CODE REVIEW

**Comment by hughsaunders** [comment]
Don't merge this until hold-until functionality is available.

**Comment by hughsaunders** [comment]
Hold-until functionality has now been merged, so this is ready for review.

File: src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java
Comment (dealako) [inline src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java]: Indentation is funny here.

File: src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java
Comment (dealako) [inline src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java]: Is this the best value for hold user? Based on the above if/else logic to get the computer, do we have access to the owner? If nothing is appropriate, then build ID is probably our best bet.

**Review by dealako — APPROVED** [review]
Looks good - minor comments.

File: src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java
Comment (hughsaunders) [reply src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java]: I think so, it could possible be the pull request author, but then I thought it would be confusing as to whether that user manually created the hold or whether it happened as part of the job.

File: src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java
Comment (hughsaunders) [reply src/main/java/com/rackspace/jenkins_nodepool/HoldStep.java]: Fixed indentation

**Pull request merged**
